pkg/reconciler/common/transformer_additional_options.go

```diff
@@ -671,3 +676,67 @@ func (ot *OptionsTransformer) createHorizontalPodAutoscalers(manifest *mf.Manife
 
 	return newHPAs, nil
 }
+
+func (ot *OptionsTransformer) updateWebhookConfiguration(u *unstructured.Unstructured) error {
+	webhookOptions, found := ot.options.WebhookConfigurationOptions[u.GetName()]
+	if !found {
+		return nil
+	}
+
+	//targetWebhookConfiguration := &admissionregistrationv1.ValidatingWebhookConfiguration{}
+	switch u.GetKind() {
+	case KindValidatingWebhookConfiguration:
+		targetWebhookConfiguration := &admissionregistrationv1.ValidatingWebhookConfiguration{}
+		err := apimachineryRuntime.DefaultUnstructuredConverter.FromUnstructured(u.Object, targetWebhookConfiguration)
+		if err != nil {
+			return err
+		}
+		for i, w := range targetWebhookConfiguration.Webhooks {
+			if u.GetName() == w.Name {
+				if webhookOptions.FailurePolicy != nil {
+					targetWebhookConfiguration.Webhooks[i].FailurePolicy = webhookOptions.FailurePolicy
+				}
+				if webhookOptions.TimeoutSeconds != nil {
+					targetWebhookConfiguration.Webhooks[i].TimeoutSeconds = webhookOptions.TimeoutSeconds
+				}
+				if webhookOptions.SideEffects != nil {
+					targetWebhookConfiguration.Webhooks[i].SideEffects = webhookOptions.SideEffects
+				}
+			}
+		}
+		// convert webhookconfigurtion to unstructured object
+		obj, err := apimachineryRuntime.DefaultUnstructuredConverter.ToUnstructured(targetWebhookConfiguration)
+		if err != nil {
+			return err
+		}
+		u.SetUnstructuredContent(obj)
+
+	case KindMutatingWebhookConfiguration:
+		targetWebhookConfiguration := &admissionregistrationv1.MutatingWebhookConfiguration{}
+		err := apimachineryRuntime.DefaultUnstructuredConverter.FromUnstructured(u.Object, targetWebhookConfiguration)
+		if err != nil {
+			return err
+		}
+		for i, w := range targetWebhookConfiguration.Webhooks {
+			if u.GetName() == w.Name {
+				if webhookOptions.FailurePolicy != nil {
+					targetWebhookConfiguration.Webhooks[i].FailurePolicy = webhookOptions.FailurePolicy
+				}
+				if webhookOptions.TimeoutSeconds != nil {
+					targetWebhookConfiguration.Webhooks[i].TimeoutSeconds = webhookOptions.TimeoutSeconds
+				}
+				if webhookOptions.SideEffects != nil {
+					targetWebhookConfiguration.Webhooks[i].SideEffects = webhookOptions.SideEffects
+				}
+			}
+		}
```

Comment:
@jkhelil Shall we extract this duplicate loop(same as line 694 to 706) outside of this function?

Reply:
I am not for doing this because, did a trial to refactor this code and it is more complicated and unreadbale, as per the validatingWebhookConfiguration and mutatingWebhookConfiguration doesnt have a common interface type in https://pkg.go.dev/k8s.io/api@v0.29.2/admissionregistration/v1
also they both have two different types validatingWebook and mutatingWebhook
so lets say we have a function that loops over webhooks in the validatingWebhookConfiguration and mutatingWebhookConfiguration, this function should take an interface type, then a switch again to identify the right type, then it should returns again an interface type of webhooks wich should be converted again.